controller/internal/routes/service_router.go

```diff
@@ -74,7 +76,27 @@ func (ir *ServiceRouter) Register(ae *env.AppEnv) {
 func (ir *ServiceRouter) List(ae *env.AppEnv, rc *response.RequestContext) {
 	// ListWithHandler won't do search limiting by logged in user
 	List(rc, func(rc *response.RequestContext, queryOptions *model.QueryOptions) (*QueryResult, error) {
-		result, err := ae.Handlers.Service.PublicQueryForIdentity(rc.Identity, rc.ApiSession.ConfigTypes, queryOptions)
+		identity := rc.Identity
+		if rc.Identity.IsAdmin {
+			if asId := rc.Request.URL.Query().Get("asIdentity"); asId != "" {
+				var err error
+				identity, err = ae.Handlers.Identity.ReadOneByQuery(fmt.Sprintf(`id = "%v" or name = "%v"`, asId, asId))
+				if err != nil {
+					return nil, err
+				}
+				if identity == nil {
+					return nil, util.NewNotFoundError("identity", "id or name", asId)
+				}
+			}
+		}
+
+		// allow overriding config types
+		configTypes := rc.ApiSession.ConfigTypes
+		if requestedConfigTypes := rc.Request.URL.Query().Get("configTypes"); requestedConfigTypes != "" {
+			configTypes = mapConfigTypeNamesToIds(ae, strings.Split(requestedConfigTypes, ","), identity.Id)
+		}
+
+		result, err := ae.Handlers.Service.PublicQueryForIdentity(identity, configTypes, queryOptions)
```

Comment:
Been wondering what `PublicQuery` means.

Reply:
PublicQuery is mean to be called for queries which come from the APIs. It checks that queries are using only local symbols and aren't chaining symbols. 

So for example when querying service, `name in ["foo", "bar"], createdAt > someDate` is ok, but `sessions.identity.name = "Admin"` would result in an error. 

Internal users would use just Query and wouldn't restricted in the symbols available to them